sql/api/src/main/scala/org/apache/spark/sql/streaming/progress.scala

```diff
@@ -258,13 +258,26 @@ class SourceProgress protected[spark] (
       ("numInputRows" -> JInt(numInputRows)) ~
       ("inputRowsPerSecond" -> safeDecimalToJValue(inputRowsPerSecond)) ~
       ("processedRowsPerSecond" -> safeDecimalToJValue(processedRowsPerSecond)) ~
-      ("metrics" -> safeMapToJValue[String](metrics, s => JString(s)))
+      ("metrics" -> safeMapToJValue[String](
+        metrics,
+        (metricsName, s) =>
+          // SPARK-53690:
```

Comment:
Technically speaking, we shouldn't have a Kafka specific handling in here and instead should build some sort of API to indicate whether we have to remove exponential format on specific metric or not.

But it's a single occurrence and Kafka is a built-in connector, so I guess it is OK. Safer approach is to check whether the source is Kafka, but maybe the metric name is not very common to be conflicted with others.